### PR TITLE
Healthcheck v2 failed

### LINKED ISSUE #13520: Broker health check with `TopicVersion.V2` failed
1、Broker health check failed with `TopicVersion.V2` but it's ok with `TopicVersion.V1`, The failed reason is:
```
2021-12-27T13:21:17,350+0800 [pulsar-io-4-71] WARN  org.apache.pulsar.client.impl.BinaryProtoLookupService - [persistent://pulsar/hidden-ip:8080/healthcheck] failed to get Partitioned metadata : {"errorMsg":"Policies not found for pulsar/hidden-ip:8080 namespace","reqId":1293596554348340272, "remote":"", "local":""}
org.apache.pulsar.client.api.PulsarClientException$BrokerMetadataException: {"errorMsg":"Policies not found for pulsar/hidden-ip:8080 namespace","reqId":1293596554348340272, "remote":"", "local":""}
        at org.apache.pulsar.client.impl.ClientCnx.getPulsarClientException(ClientCnx.java:1154) ~[org.apache.pulsar-pulsar-client-original-2.10.0-SNAPSHOT.jar:2.10.0-SNAPSHOT]
        at org.apache.pulsar.client.impl.ClientCnx.handlePartitionResponse(ClientCnx.java:607) [org.apache.pulsar-pulsar-client-original-2.10.0-SNAPSHOT.jar:2.10.0-SNAPSHOT]
        at org.apache.pulsar.common.protocol.PulsarDecoder.channelRead(PulsarDecoder.java:130) [org.apache.pulsar-pulsar
```
Policies not found for namespace  `pulsar/host:8080` under  znode `admin/policies`. 

But Why v1 is successful?  v1's healthcheck topic name is global, `checkReplication` is skipped.

I cannot decide the best solution to fix it.

### PR DESCRIPTION
**Pull request by gaozhangmin**
fix #13520 
### Documentation

Check the box below and label this PR (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by liudezhi2098 — COMMENTED** [review]
can you add a test?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 1]
```diff
@@ -1427,7 +1428,7 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
     @VisibleForTesting
     public CompletableFuture<List<String>> getReplicationClusters(TopicName topicName) {
         CompletableFuture<Optional<TopicPolicies>> future = new CompletableFuture<>();
-        if (isSystemTopic()) {
+        if (isSystemTopic() || HEARTBEAT_TENANT.equals(topicName.getTenant())) {
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:1431]: Can we include heartbeat topic in SystemTopic?

Comment (gaozhangmin) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java]: done @Jason918

**Comment by gaozhangmin** [comment]
@liudezhi2098  The test is already covered by `org.apache.pulsar.broker.admin.AdminApiHealthCheckTest#testHealthCheckupV2Error`
The previous unit test `testHealthCheckupV2 `  created a heartbeat namespace Manually before running health check.
But, In fact, the heartbeat namespace would not be created  in cluster.

**Review by eolivelli — CHANGES_REQUESTED** [review]
This PR looks like adding some tweaks specific for this special topic.

Did you check if there is a better way?
(I am sorry I don't have time today to send you a suggestion)

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 2]
```diff
@@ -2955,7 +2956,7 @@ public CompactedTopic getCompactedTopic() {
 
     @Override
     public boolean isSystemTopic() {
-        return false;
+        return HEARTBEAT_TENANT.equals(TopicName.get(topic).getTenant());
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:2959]: Not like this. There are some other system topics.
The origin check seems to be `org.apache.pulsar.broker.service.BrokerService#isSystemTopic`.

**Review by Jason918 — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java (hunk 2 above)
Comment (gaozhangmin) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java]: #13495 made changes on `getReplicationClusters`  don't need this change any more

**Comment by gaozhangmin** [comment]
> This PR looks like adding some tweaks specific for this special topic.

@eolivelli  The direct reason  is  that heartbeat namespace was not created. just registered by `registerBootstrapNamespaces` during startup. The optional fix maybe, we can create this heartbeat namespace during startup.

**Comment by massakam** [comment]
There may also be namespaces under the `pulsar` tenant that are not for healthcheck. For example, it is possible for a user to create a namespace named `pulsar/test`. For such namespaces, we should not skip these checks.

**Comment by gaozhangmin** [comment]
> There may also be namespaces under the `pulsar` tenant that are not for healthcheck. For example, it is possible for a user to create a namespace named `pulsar/test`. For such namespaces, we should not skip these checks.

Fixed in the last commit.

**Comment by massakam** [comment]
I think `NonPersistentTopic.java` should be modified as well as `PersistentTopic.java`.
https://github.com/apache/pulsar/blob/ef42af2803e0d8b8141331807a0549c4d3277361/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java#L521-L525

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiHealthCheckTest.java [hunk 3]
```diff
@@ -66,18 +63,7 @@ public void testHealthCheckupV1() throws Exception {
         admin.brokers().healthcheck(TopicVersion.V1);
     }
 
-    @Test(expectedExceptions = PulsarAdminException.class)
-    public void testHealthCheckupV2Error() throws Exception {
-        admin.brokers().healthcheck(TopicVersion.V2);
-    }
-
-    @Test
```
Comment (massakam) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiHealthCheckTest.java:74]: `@Test` annotation is missing.

**Review by massakam — APPROVED** [review]
LGTM

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 4]
```diff
@@ -1371,6 +1373,12 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
         if (!name.isGlobal()) {
             return CompletableFuture.completedFuture(null);
         }
+        NamespaceName heartbeatNamespace = NamespaceService.getHeartbeatNamespaceV2(
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:1376]: we  move this into the PulsarService as a field, this way  we don't have to create this every time

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by Jason918 — APPROVED** [review]
LGTM

**Comment by gaoran10** [comment]
The https://github.com/apache/pulsar/pull/14658 cherry-pick this PR to `branch-2.9`.

**Pull request merged**